NCS.DSS.Transfer/GetTransferHttpTrigger/Function/GetTransferHttpTrigger.cs

```diff
@@ -15,71 +15,99 @@ namespace NCS.DSS.Transfer.GetTransferHttpTrigger.Function
-                _logger.LogInformation("Unable to locate 'TouchpointId' in request header.");
+                _logger.LogWarning("Unable to locate 'TouchpointId' in request header");
                 return new BadRequestResult();
             }
 
-            _logger.LogInformation("Get Transfers C# HTTP trigger function processed a request. " + touchpointId);
-
             if (!Guid.TryParse(customerId, out var customerGuid))
+            {
+                _logger.LogWarning("Unable to locate 'CustomerId' in request header");
                 return new BadRequestObjectResult(customerGuid.ToString());
+            }
 
             if (!Guid.TryParse(interactionId, out var interactionGuid))
+            {
+                _logger.LogWarning("Unable to locate 'InteractionId' in request header");
                 return new BadRequestObjectResult(interactionGuid.ToString());
+            }
+
+            _logger.LogInformation("Header validation has succeeded. Touchpoint ID: {TouchpointId}", touchpointId);
 
+            _logger.LogInformation("Attempting to see if customer exists. Customer GUID: {CustomerGuid}", customerGuid);
             var doesCustomerExist = await _resourceHelper.DoesCustomerExist(customerGuid);
 
             if (!doesCustomerExist)
+            {
+                _logger.LogWarning("Customer does not exist. Customer GUID: {CustomerGuid}", customerGuid);
                 return new NoContentResult();
+            }
+            _logger.LogInformation("Customer exists. Customer GUID: {CustomerGuid}", customerGuid);
 
-            var doesInteractionExist = _resourceHelper.DoesInteractionResourceExistAndBelongToCustomer(interactionGuid, customerGuid);
+            _logger.LogInformation("Attempting to check Interaction exists for customer. Customer GUID: {CustomerGuid}", customerGuid);
+            var doesInteractionExist = await _resourceHelper.DoesInteractionResourceExistAndBelongToCustomer(interactionGuid, customerGuid);
 
             if (!doesInteractionExist)
+            {
+                _logger.LogWarning("Interaction does not exist for customer. Customer GUID: {CustomerGuid}. Interaction GUID: {InteractionGuid}", customerGuid, interactionGuid);
                 return new NoContentResult();
-
+            }
+            _logger.LogInformation("Interaction exists for customer. Customer GUID: {CustomerGuid}. Interaction GUID: {InteractionGuid}", customerGuid, interactionGuid);
+            
+            _logger.LogInformation("Attempting to retrieve Transfer(s) for customer. Customer GUID: {CustomerGuid}", customerGuid);
             var transfers = await _transferGetService.GetTransfersAsync(customerGuid);
 
             if (transfers == null)
+            {
+                _logger.LogInformation("Transfer does not exist for customer. Customer GUID: {CustomerGuid}", customerGuid);
+                _logger.LogInformation("Function {FunctionName} has finished invoking", nameof(GetTransferHttpTrigger));
```

Comment:
This "Function has finished invoking method" isnt necessary here as you are already returning a bad result. This same line needs removing for every other trigger also

Reply:
Kept the log line as the entire function has completely successfully and returning null for failing to retrieve a transfer that doesn't exist, is successful. Thus the code returns a successful 204 instead of a 400 error.